sam.c

```diff
@@ -1702,7 +1702,7 @@ static bool cmd_write(Vis *vis, Win *win, Command *cmd, const char *argv[], Sele
 			file_name_set(file, path);
 			same_file = true;
 		}
-		if (same_file)
+		if (same_file || !existing_file)
```

Comment:
While this fixes the case of opening an non-existing file, there is another issue illustrated by:

    $ vis exisiting-file
    :w non-existing-file
    :w

In this case we update `file->stat` with the data of the wrong file, resulting in a warning when attempting to write the one initially opened.

I will shortly push a change with:

    if (same_file || (!existing_file && strcmp(file->name, path) == 0))

which should hopefully fix this.

Reply:
Thanks!